Sources/SwiftProtobuf/TextFormatDecoder.swift

```diff
@@ -568,14 +563,20 @@ internal struct TextFormatDecoder: Decoder {
                     throw TextFormatDecodingError.malformedText
                 }
             }
-            if let key = try scanner.nextKey() {
+            if let key = try scanner.nextKey(allowExtensions: ignoreExtensionFields) {
                 switch key {
                 case "key", "1":
                     try KeyType.decodeSingular(value: &keyField, from: &self)
                 case "value", "2":
                     try ValueType.decodeSingular(value: &valueField, from: &self)
                 default:
-                    throw TextFormatDecodingError.unknownField
+                    if ignoreExtensionFields && key.hasPrefix("[") {
```

Comment:
Can we ever legally see an unknown field or extension field inside a _map_?   The map encoding is pretty tightly specified.

Reply:
Yea, I don't think you should, but since anyone can hand code one, the question becomes what should we do?  As far as I can tell, with the C++ code, if you turn on their options, they too will skip these fields.

(And it seems sorta odd if one enabling ignoring unknown fields that you'd ever actually get an unknownField error)

Reply:
Here's the spec part for map - https://protobuf.dev/reference/protobuf/textformat-spec/#map, it's basically just a repeated message field, so I went with the same sorta support there for unknown fields.

Reply:
That makes sense.  I suppose we'll need test cases for maps with extensions in various places?

Reply:
Yup, the changes to Tests/SwiftProtobufTests/Test_TextFormat_Map_proto3.swift adds that, testing fields and extensions independently to be sure things fail/pass as expected.